docs/library/bluetooth.rst

```diff
@@ -0,0 +1,306 @@
+
+In order to save space in the firmware, these constants are not included on the
+``bluetooth`` module. Add the ones that you need from the list above to your
+program.
+
+
+Broadcaster Role (Advertiser)
+-----------------------------
+
+.. method:: BLE.gap_advertise(interval_ms, adv_data=None, resp_data=None, connectable=True)
+
+    Starts advertising at the specified interval. To stop advertising, set
+    interval_ms to ``0`` or ``None``.
+
+    ``adv_data`` and ``resp_data`` can be any type that implements the buffer
+    protocol (e.g. ``bytes``, ``bytearray``, ``str``).
+
+
+Observer Role (Scanner)
+-----------------------
+
+.. method:: BLE.gap_scan(duration_ms, [interval_ms], [window_ms])
+
+    Run a scan operation lasting for the specified duration.
+
+    To scan indefinitely, set ``duration_ms`` to ``0``.
+
+    To stop scanning, set ``duration_ms`` to ``None``.
+
+    Use ``interval_ms`` and ``window_ms`` to optionally configure the duty
+    cycle. The scanner will run for ``window_ms`` every ``interval_ms`` for a
+    total of ``duration_ms`` milliseconds. The default interval and window are
+    1.28 seconds and 11.25 milliseconds respectively (background scanning).
+
+    For each scan result, the ``_IRQ_SCAN_RESULT`` event will be raised.
+
+    When scanning is stopped (either due to the duration finishing or when
+    explicitly stopped), the ``_IRQ_SCAN_COMPLETE`` event will be raised.
+
+
+Peripheral Role (GATT Server)
+-----------------------------
+
+A BLE peripheral has a set of registered services. Each service may contain
+characteristics, which each have a value. Characteristics can also contain
+descriptors, which themselves have values.
+
+These values are stored locally and can be read from or written to by a remote
+central device. Additionally, a peripheral can "notify" its value to a connected
+central via its connection handle.
+
+.. method:: BLE.gatts_register_services(services_definition)
+
+    Configures the peripheral with the specified services, replacing any
+    existing services.
+
+    ``services_definition`` is a list of *services*, where each *service* is a
+    two-element tuple containing a UUID and a list of *characteristics*.
+
+    Each *characteristic* is a two-or-three-element tuple containing a UUID, a
+    *flags* value, and optionally a list of *descriptors*.
+
+    Each *descriptor* is a two-element tuple containing a UUID and a flags value.
+
+    The *flags* are a bitwise-OR combination of the ``FLAGS_READ``,
+    ``FLAGS_WRITE`` and ``FLAGS_NOTIFY`` values defined below.
+
+    The return value is a list (one element per service) of tuples (each element
+    is a value handle). Characteristics and descriptor handles are flattened
+    into the same tuple, in the order that they are defined.
+
+    The following example that registers two services (Heart Rate, and Nordic
+    UART)::
+
+        HR_SERVICE = bluetooth.UUID(0x180D)
```

Comment:
Wouldn't this be the HR_UUID, and then `HR_SERVICE=(HR_UUID, (HR_CHAR,))`?  I think this would be a cleaner way to present these items.  Then it'd be just `bt.gatts_register_services((HR_SERVICE, UART_SERVICE))`.